cmd/cvc-operator/controller/cstorvolumeclaim.go

```diff
@@ -521,6 +529,92 @@ func randomizePoolList(list *apis.CStorPoolInstanceList) *apis.CStorPoolInstance
 	for _, randomIdx := range perm {
 		res.Items = append(res.Items, list.Items[randomIdx])
 	}
-
 	return res
 }
+
+// getOrCreatePodDisruptionBudget will does following things
+// 1. It tries to get the PDB that was created among volume replica pools.
+// 2. If PDB exist it returns the PDB.
+// 3. If PDB doesn't exist it creates new PDB(With CSPC hash)
+func getOrCreatePodDisruptionBudget(
+	cvObj *apis.CStorVolume, cspcName string) (*policy.PodDisruptionBudget, error) {
+	poolNames, err := cvr.GetReplicaPoolNames(cvObj)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"failed to get volume replica pool names of volume %s",
+			cvObj.Name)
+	}
+	pdbLabels := cvclaim.GetPDBPoolLabels(poolNames)
+	labelSelector := apispdb.GetPDBLabelSelector(pdbLabels)
+	pdbList, err := apispdb.KubeClient().
+		WithNamespace(getNamespace()).
+		List(metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"failed to list PDB belongs to pools %v", pdbLabels)
+	}
+	if len(pdbList.Items) > 1 {
+		return nil, errors.Wrapf(err,
+			"current PDB count %d of pools %v",
+			len(pdbList.Items),
+			pdbLabels)
+	}
+	if len(pdbList.Items) == 1 {
+		return &pdbList.Items[0], nil
+	}
+	return createPDB(cvObj, poolNames, cspcName)
+}
+
+// createPDB creates PDB for cStorVolumes based on arguments
+func createPDB(cvObj *apis.CStorVolume,
+	poolNames []string, cspcName string) (*policy.PodDisruptionBudget, error) {
+	// Calculate minAvailable value from cStorVolume replica count
+	//minAvailable := (cvObj.Spec.ReplicationFactor >> 1) + 1
+	maxUnavailableIntStr := intstr.FromInt(1)
+
+	//build podDisruptionBudget for volume
+	pdbObj := policy.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: cspcName,
+			Labels:       cvclaim.GetPDBLabels(poolNames, cspcName),
+		},
+		Spec: policy.PodDisruptionBudgetSpec{
+			MaxUnavailable: &maxUnavailableIntStr,
+			Selector:       getPDBSelector(poolNames),
+		},
+	}
+	// Create podDisruptionBudget
+	return apispdb.KubeClient().
+		WithNamespace(cvObj.Namespace).
```

Comment:
we are creating in cvObj, but, listing in getNamespace().. can they be different?

Reply:
Always both will be the same but to maintain consistency I will update to it to getNamesapce()...